tools/map/build_junguozhi.py

```
@@ -0,0 +1,480 @@
+    segs = read_segments()
+    lex = county_lexicon()
+    cnty_xy = chgis_points('cnty')
+    pref_xy = chgis_points('pref')
+    # 郡國志-CHGIS 이체자. 한 글자 차이로 같은 郡이 남남이 되고, 앵커를 못 찾은 郡은
+    # 아래 필터가 통째로 꺼져 동명이인을 1500km 밖에서 물어온다.
+    VARIANT = str.maketrans('鴈郁閒雒沇涼竝', '雁鬱間洛兗凉并')
+    # CHGIS 판도 밖이라 앵커가 있을 수 없는 郡(交趾·九真·日南·樂浪·帶方…).
+    # 좌표는 external-places.json 이 사료·Wikidata 로 비정한 것이며 필터 기준점으로만 쓴다.
+    EXTRA_ANCHOR = {}
+    try:
+        for _e in json.load(open('data/map/external-places.json'))['places']:
```

Ruff (0.16.1)

[warning] Use a context manager for opening files

(SIM115)